src/main/java/com/blogify/blogapi/service/event/PostService.java

```diff
@@ -0,0 +1,22 @@
+package com.blogify.blogapi.service.event;
+
+import com.blogify.blogapi.model.Post;
+import com.blogify.blogapi.repository.PostRepository;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@AllArgsConstructor
+public class PostService{
+    private final PostRepository postRepository;
+
+    public List<Post> getAll(){
+        return postRepository.findAll();
+    }
+    public Optional<Post> getById(String id){
+        return postRepository.findById(id);
```

Comment:
L5 - should not return Optional<T> , eg: Optional<Post> --> Post
manage each error with an Exception handle if necessary.